x-pack/plugins/endpoint/server/routes/endpoints.ts

```diff
@@ -51,6 +54,33 @@ export function registerEndpointRoutes(router: IRouter, endpointAppContext: Endp
       }
     }
   );
+
+  router.get(
+    {
+      path: '/api/endpoint/endpoints/{id}',
+      validate: {
+        params: schema.object({ id: schema.string() }),
+      },
+      options: { authRequired: true },
+    },
+    async (context, req, res) => {
+      try {
+        const query = kibanaRequestToEndpointFetchQuery(req, endpointAppContext);
+        const response = (await context.core.elasticsearch.dataClient.callAsCurrentUser(
+          'search',
+          query
+        )) as SearchResponse<EndpointMetadata>;
+
+        if (response.hits.hits.length === 0) {
+          return res.notFound({ body: 'Endpoint Not Found' });
```

Comment:
Q. Do we localize these message from the server? (cc/ @nnamdifrankie )

Reply:
I think we should just return a 404 `res.notFound()`, and this will let the front end localize or internationalize the presentation.

Reply:
the source for `notFound` just does this: `options.body || 'Not Found'`  So it's not i18n as it stands, and the front-end can freely ignore the body as it wants

Reply:
👍